Library/Homebrew/requirements/gfortran_requirement.rb

```diff
@@ -0,0 +1,33 @@
+class GfortranRequirement < Requirement
+  fatal true
+  default_formula "gcc"
+
+  satisfy build_env: false do
+    next false unless which "gfortran"
+    next true unless @version
+    gfortran_version = Utils.popen_read("gfortran", "--version").split("\n").first.split(" ").last.chomp
```

Comment:
I think we can grab some of this logic and insert it into https://github.com/Homebrew/homebrew-core/pull/8790 perhaps after we merge a decent enough first cut?